org/postgresql/jdbc2/TypeInfoCache.java

```diff
@@ -150,9 +153,14 @@ public synchronized void addCoreType(String pgTypeName, Integer oid, Integer sql
         Character delim = ',';
         _arrayOidToDelimiter.put(oid, delim);
 
-        String pgArrayTypeName = "_" + pgTypeName;
```

Comment:
> It would be better if information_schema was used instead. It doesn't suffer from _ prefix issue.

`select udt_name from information_schema.columns where table_schema='public'` returns `_int4`. There is no column with value `int4[]`.

>a better solution would be to cache all arrays instead

Referenced issue is about replacing cache value with alias. AFAIU it is irrelevant whether to prepopulate cache or not.